### DISCUSSION TITLE
Can I be anything other than Host?

### DISCUSSION THREAD

**Discussion by skintigh** [pyusb/pyusb#491] [Q&A] [answered]
I assume the answer is no, but can my program (running on a Windows PC) be anything other than the USB host? I'd like to emulate a device, like a mass storage device, and connect to another machine running as a host. So far the answer seems to be "no you can't, not without rewriting your OS's USB drivers."

**Comment by mcuee** [comment, marked as answer]
No you cannot. pyusb is based on libusb, and libusb only works in the host side.

You can look at Linux specific libraries like the following, from the author of python-libusb1.
https://github.com/vpelletier/python-functionfs
